src/main/java/com/cleanroommc/groovyscript/compat/mods/pyrotech/Worktable.java

```diff
@@ -0,0 +1,500 @@
+    public void replaceShapeless(ResourceLocation name, ItemStack output, IIngredient tool, int damage, List<IIngredient> input) {
+        shapelessBuilder()
+                .tool(tool, damage)
+                .input(input)
+                .output(output)
+                .name(name)
+                .replaceByName()
+                .register();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.ADDITION, description = "groovyscript.wiki.minecraft.crafting.replaceShaped0", example = @Example(value = "item('minecraft:chest'), item('minecraft:iron_axe') | item('minecraft:stone_axe'), 3, [[ore('logWood'),ore('logWood'),ore('logWood')],[ore('logWood'),null,ore('logWood')],[ore('logWood'),ore('logWood'),ore('logWood')]]", commented = true))
+    public void replaceShaped(ItemStack output, IIngredient tool, int damage, List<List<IIngredient>> input) {
+        shapedBuilder()
+                .tool(tool, damage)
+                .matrix(input)
+                .output(output)
+                .replace()
+                .register();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.ADDITION, description = "groovyscript.wiki.minecraft.crafting.replaceShaped1", example = @Example(value = "'gold_to_diamonds', item('minecraft:diamond') * 8, item('minecraft:diamond_pickaxe'), 4, [[item('minecraft:gold_ingot'),item('minecraft:gold_ingot'),item('minecraft:gold_ingot')],[item('minecraft:gold_ingot'),null,item('minecraft:gold_ingot')],[item('minecraft:gold_ingot'),item('minecraft:gold_ingot'),item('minecraft:gold_ingot')]]", commented = true))
+    public void replaceShaped(String name, ItemStack output, IIngredient tool, int damage, List<List<IIngredient>> input) {
+        shapedBuilder()
+                .tool(tool, damage)
+                .matrix(input)
+                .output(output)
+                .name(name)
+                .replaceByName()
+                .register();
+    }
+
+    @MethodDescription(type = MethodDescription.Type.ADDITION, description = "groovyscript.wiki.minecraft.crafting.replaceShaped1", example = @Example(value = "resource('minecraft:sea_lantern'), item('minecraft:diamond_pickaxe'), 3, item('minecraft:clay'), [[item('minecraft:glowstone')],[item('minecraft:glowstone')],[item('minecraft:glowstone')]]", commented = true))
+    public void replaceShaped(ResourceLocation name, ItemStack output, IIngredient tool, int damage, List<List<IIngredient>> input) {
+        shapedBuilder()
+                .tool(tool, damage)
+                .matrix(input)
+                .output(output)
+                .name(name)
+                .replaceByName()
+                .register();
+    }
+
+    @MethodDescription(example = @Example("item('pyrotech:iron_hunters_knife')"))
+    public void removeByOutput(IIngredient output) {
+        removeByOutput(output, true);
+    }
+
+    public void removeByOutput(IIngredient output, boolean log) {
```

Comment:
annotate `@GroovyBlacklist`, and the other methods you dont want being callable from groovyscript